### PR TITLE
chore: add helm charts to deploy mayastor control plane

### PR DESCRIPTION
**Pull request by blaisedias**
Create helm charts under chart directory with profiles
 - develop
 - release
 - test
Create script to generate helm charts similar to that present in
the mayastor repository
Update the yaml files in deploy to the generated versions with profile develop

### CODE REVIEW

File: chart/templates/core-agents-deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    matchLabels:
+      app: core-agents
+  template:
+    metadata:
+      labels:
+        app: core-agents
+    spec:
+      containers:
+        - name: core
+          image: {{ .Values.mayastorCP.registry }}mayadata/mcp-core:{{ .Values.mayastorCP.tag}}
```
Comment (paulyoong) [inline chart/templates/core-agents-deployment.yaml:20]: Should the `mayadata` portion be incorporated as part of the {{ .Values.mayastorCP.registry }}?

File: chart/templates/rest-deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+      containers:
+        - name: rest
+          image: {{ .Values.mayastorCP.registry }}mayadata/mcp-rest:{{ .Values.mayastorCP.tag}}
+          imagePullPolicy: {{ .Values.mayastorCP.pullPolicy }}
+          args:
+            - "--dummy-certificates"
+            - "--no-auth"
+            - "-nnats"
+            - "--http=0.0.0.0:8081"
+            - "--timeout=6s"
```
Comment (paulyoong) [inline chart/templates/rest-deployment.yaml]: This now needs to be changed to `request-timeout`.

**Review by paulyoong — APPROVED** [review]

File: chart/templates/rest-deployment.yaml (hunk 2 above)
Comment (paulyoong) [reply chart/templates/rest-deployment.yaml]: I saw you just changed this so ignore my comment.

File: chart/templates/core-agents-deployment.yaml (hunk 1 above)
Comment (blaisedias) [reply chart/templates/core-agents-deployment.yaml:20]: I don't think so that part in invariant, on dockerhub the image name is `mayadata/mcp-core`

**Comment by blaisedias** [comment]
bors merge

File: chart/templates/core-agents-deployment.yaml (hunk 1 above)
Comment (paulyoong) [reply chart/templates/core-agents-deployment.yaml:20]: Fair enough

**Comment by blaisedias** [comment]
bors merge

**Pull request merged**